### ISSUE TITLE
[Suggest] option for python path in preference

### ISSUE THREAD

**Issue by alvis** [shd101wyy/markdown-preview-enhanced#683]
Python has an interesting ecosystem. Instead of the default global install from the OS, it is usually used within a virtual environment managed by anaconda or alike.

The consequence is that the path of the python binary from these environments is not usually attached to $PATH, and hence the code in the chunk cannot find the right packages.

Let me give an example case. I use MacOS and the default python version is 2.7. In order to use python 3 syntax, I have to specify `cmd=/Users/{user}/anaconda3/bin/python` for all of my python code chunk. It's annoying, and more importantly, it is not portable. Some may suggest attaching the directory to $PATH, but it doesn't work because some system tools I have do use python 2.7. The change will break these tools.

Suggested solution: add a configuration in the preference pane for the default python executable used for code chunk. How does it sound to everyone? I am willing to make a PR for that if people like the idea.

**Comment by specter119** [comment]
@alvis I'm not the author, but I have some suggestions for you.

1. use `pyenv`, you can easily install  `minicanda`by `pyenv`. It can solve most cases when you work with console, and also work for MPE if you run chunk with `{cmd=True}`.

1. use local setting for specific folder(project) (adapt `python.pythonPath` in `.vscode/settings.json`), this will help you handle all cases with python in your vscode.

I tried atom for a period of time but turn to vscode now, I'm not sure whether atom can also set some local settings to override global settings like vscode. but at least you can use `pyenv` for this plugin.

**Comment by alvis** [comment]
@specter119: Thanks for your suggestion. Sadly, it doesn't seem to work with atom.

The currently implementation is that MPE looks for the executable from `$PATH` as
https://github.com/shd101wyy/markdown-preview-enhanced/blob/b17dc5b12188e7cce904d0dce829d23cad2bda7e/lib/code-chunk.coffee#L147

Therefore, without the ability to modify `cmd` or `$PATH`, MPE can't pick the right executable. I don't use vscode, but I suspect why it works with specifying the path in `python.pythonPath` because it attaches the path to `$PATH` in the background.

**Comment by specter119** [comment]
@alvis Emmm,  but `pyenv` really changes the `python` in $PATH depending on your $PWD. I think you may have a try.

**Comment by alvis** [comment]
@specter119 you're right that pyenv fix $PATH in console. However, the environment cannot be picked up by atom since it is run under different shell. Therefore MPE would only pick up the default $PATH, not the new $PATH patched by pyenv.

A workaround in this case would be launching atom from the console after $PATH is patched, so that atom can pick up $PATH from the shell. However, it is far from ideal.
